File: frontend/components/opportunities/opportunityStages.ts
export const OPPORTUNITY_STAGE_ORDER = [
  "lead",
  "qualified",
  "proposal",
  "negotiation",
  "closed_won",
  "closed_lost",
] as const;

export const OPPORTUNITY_STAGE_LABELS: Record<string, string> = {
  lead: "Lead",
  qualified: "Qualified",
  proposal: "Proposal",
  negotiation: "Negotiation",
  closed_won: "Closed Won",
  closed_lost: "Closed Lost",
};

import { getOpportunityStageStyle as getCentralOpportunityStageStyle } from "@/lib/statusStyles";

export function normalizeOpportunityStage(stage?: string | null) {
  return (stage ?? "").toLowerCase().replace(/\s+/g, "_");
}

export function getOpportunityStageLabel(stage?: string | null) {
  const key = normalizeOpportunityStage(stage);
  return OPPORTUNITY_STAGE_LABELS[key] ?? stage ?? "Unstaged";
}

export function getOpportunityStageStyle(stage?: string | null) {
  return getCentralOpportunityStageStyle(normalizeOpportunityStage(stage) || "unstaged");
}

File: backend/app/modules/sales/opportunity_stages.py
from __future__ import annotations

OPPORTUNITY_STAGE_ORDER = [
    "lead",
    "qualified",
    "proposal",
    "negotiation",
    "closed_won",
    "closed_lost",
]

OPPORTUNITY_STAGE_LABELS = {
    "lead": "Lead",
    "qualified": "Qualified",
    "proposal": "Proposal",
    "negotiation": "Negotiation",
    "closed_won": "Closed Won",
    "closed_lost": "Closed Lost",
    "unstaged": "Unstaged",
}

OPPORTUNITY_STAGE_SET = set(OPPORTUNITY_STAGE_ORDER)
OPPORTUNITY_CLOSED_STAGE_SET = {"closed_won", "closed_lost"}
OPPORTUNITY_STAGE_PATTERN = f"^({'|'.join(OPPORTUNITY_STAGE_ORDER)})$"
OPPORTUNITY_STAGE_CHECK_SQL = (
    "sales_stage IS NULL OR sales_stage IN ("
    + ", ".join(f"'{stage}'" for stage in OPPORTUNITY_STAGE_ORDER)
    + ")"
)
